src/providers/aws.rs

```diff
@@ -1289,40 +1308,14 @@ impl RegionLauncher {
                     tracing::trace!("retrying instance termination");
                     continue;
                 } else {
-                    tracing::warn!("failed to terminate tsunami instances: {}", e);
-                    let e = Err(e).context("failed to terminate tsunami instances");
-                    res = match res {
-                        Ok(_) => e,
-                        Err(r) => Err(r.wrap_err(e.unwrap_err())),
-                    };
-                    break;
-                }
-            }
-        }
-
-        tracing::debug!("cleaning up temporary resources");
-        if !self.ssh_key_name.trim().is_empty() {
-            let key_span = tracing::trace_span!("key", name = %self.ssh_key_name);
-            let r = async {
-                tracing::trace!("removing keypair");
-                let mut req = rusoto_ec2::DeleteKeyPairRequest::default();
-                req.key_name = self.ssh_key_name.clone();
-                if let Err(e) = client.delete_key_pair(req).await {
-                    tracing::warn!("failed to clean up temporary SSH key: {}", e);
-                    Err(e).context("failed to clean up temporary SSH key")?;
+                    // Why is `?` here ok? either:
+                    // 1. there was no spot capacity. So self.instances will be empty, and this
+                    //    block will get skipped, so sg will get cleaned up below.
+                    // 2. there were instances, but we couldn't terminate them. Then, the sg will
+                    //    still be attached to them, so there's no point trying to delete it.
+                    Err(e).context("failed to terminate tsunami instances")?;
```

Comment:
I think `eyre` favors `wrap_err` over `context` (though I _think_ they both do the same thing).